src/Microsoft.AspNet.Http/Features/FormFeature.cs

```diff
@@ -67,16 +68,6 @@ public bool HasFormContentType
 
         public IFormCollection ReadForm()
         {
-            if (Form != null)
```

Comment:
Which can just as happily also call: ```ReadFormAsync().GetAwaiter().GetResult()``` or ```ReadFormAsync().Result()``` but it doesn't come with the incorrect assumption that the synchronous method is not [Sync over Async](http://blogs.msdn.com/b/pfxteam/archive/2012/04/13/10293638.aspx)?